spec/system/candidate_interface/candidate_new_user_with_course_params_spec.rb

```diff
@@ -0,0 +1,123 @@
+    and_the_course_i_selected_only_has_one_site
+
+    when_i_arrive_from_find_to_a_course_that_is_open_on_apply
+    and_i_click_apply_on_apply
+    then_the_url_should_contain_the_course_code_and_provider_code_param
+    when_i_fill_in_the_eligiblity_form_with_yes
+    then_the_url_should_contain_the_course_code_and_provider_code_param
+
+    when_i_submit_my_email_address
+    and_click_on_the_magic_link
+    then_i_should_see_the_courses_review_page
+    and_i_should_see_the_course_name_and_code
+    and_i_should_see_the_site(@site)
+    and_my_find_from_id_course_should_be_set_to_nil
+
+    given_the_course_i_selected_has_multiple_sites
+
+    when_i_submit_my_email_address
+    and_click_on_the_magic_link
+    then_i_should_see_the_course_choices_site_page
+    and_i_should_see_the_site(@site1)
+    and_i_should_see_the_site(@site2)
+    and_my_find_from_id_course_should_be_set_to_nil
+  end
+
+  def and_confirm_course_choice_from_find_is_activated
+    FeatureFlag.activate('confirm_course_choice_from_find')
+  end
+
+  def given_the_pilot_is_open
+    FeatureFlag.activate('pilot_open')
+  end
+
+  def and_the_course_i_selected_only_has_one_site
+    @course = create(:course, exposed_in_find: true, open_on_apply: true, name: 'Potions')
+    @site = create(:site, provider: @course.provider)
+    create(:course_option, site: @site, course: @course, vacancy_status: 'B')
+  end
+
+  def given_the_course_i_selected_has_multiple_sites
+    @course_with_multiple_sites = create(:course, exposed_in_find: true, open_on_apply: true, name: 'Herbology')
+    @site1 = create(:site, provider: @course_with_multiple_sites.provider)
+    @site2 = create(:site, provider: @course_with_multiple_sites.provider)
+    create(:course_option, site: @site1, course: @course_with_multiple_sites, vacancy_status: 'B')
+    create(:course_option, site: @site2, course: @course_with_multiple_sites, vacancy_status: 'B')
+
+    visit candidate_interface_sign_up_path providerCode: @course_with_multiple_sites.provider.code, courseCode: @course_with_multiple_sites.code
+  end
+
+  def when_i_arrive_from_find_to_a_course_that_is_open_on_apply
+    visit candidate_interface_apply_from_find_path providerCode: @course.provider.code, courseCode: @course.code
+  end
+
+  def and_i_click_apply_on_apply
+    click_on t('apply_from_find.apply_button')
+  end
+
+  def then_the_url_should_contain_the_course_code_and_provider_code_param
+    expect(page.current_url).to have_content "providerCode=#{@course.provider.code}"
+    expect(page.current_url).to have_content "courseCode=#{@course.code}"
+  end
+
+  def when_i_fill_in_the_eligiblity_form_with_yes
+    within_fieldset('Are you a citizen of the UK, EU or EEA?') do
+      choose 'Yes'
+    end
+
+    within_fieldset('Did you gain all your qualifications at institutions based in the UK?') do
+      choose 'Yes'
+    end
+
+    click_on 'Continue'
+  end
+
+  def when_i_submit_my_email_address
+    perform_enqueued_jobs do
```

Comment:
This had to be added as the magic link uses an async call on sign up. This was causing `open_email(@email)` to return nil. Interestingly this did not occur on other specs which use the same service. Adding the query string seems to be the case 🤷‍♂

Reply:
You can use `sidekiq: true` to run the jobs inline:

https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/blob/c136e7c134859301a7105ece79d974606d403a09/spec/system/candidate_interface/candidate_accepts_offer_spec.rb#L3

Reply:
I actually tried this earlier and still got the following error

![image](https://user-images.githubusercontent.com/42515961/72835156-1d1bd600-3c82-11ea-91f0-1aeb90156547.png)

Reply:
That's awesome - a mystery! Lets investigate tomorrow.

Reply:
Confirmed this on my machine, I don't really understand why but fine for now!